File: src/main/java/pl/exsio/querydsl/entityql/examples/Application.java
package pl.exsio.querydsl.entityql.examples;

import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

import java.util.List;

@Configuration
@ComponentScan("pl.exsio.querydsl.entityql.examples")
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @Bean
    public CommandLineRunner commandLineRunner(List<Example> examples) {
        return args -> examples.forEach(Example::run);
    }
}

File: kotlin-gradle-jpa-dynamic/src/main/kotlin/pl/exsio/querydsl/entityql/examples/Application.kt
package pl.exsio.querydsl.entityql.examples

import org.springframework.boot.CommandLineRunner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration


@SpringBootApplication
@Configuration
@ComponentScan("pl.exsio.querydsl.entityql.examples")
open class Application {

    @Bean
    open fun commandLineRunner(examples: List<Example>): CommandLineRunner {
        return object : CommandLineRunner {
            override fun run(vararg args: String?) {
                examples.forEach { example ->
                    println("Running Example: " + example::class.qualifiedName)
                    example.run()
                }
            }
        }
    }
}

fun main(args: Array<String>) {
    runApplication<Application>(*args);
}
